docs: DeepBook/DeFi builder path (#27434)
MystenLabs/sui: docs/content/onchain-finance/deepbook/deepbookv3-sdk/ptb-cli-cookbook.mdx

---

> title: PTB and CLI Cookbook for DeepBook

Reviewer: ```suggestion
title: Transaction and CLI Cookbook for DeepBook
```

Reviewer: if new users don't know what PTB stands for, might be confusing as a title

> sidebar_label: PTB and CLI Cookbook

Reviewer: ```suggestion
sidebar_label: Transaction and CLI Cookbook
```

    Reader can build common DeepBook PTBs from copy-paste CLI and TypeScript
    templates and avoid the most common CLI syntax errors.
  requires:
    - has_frontmatter:
        - title
        - description
        - keywords
      label: Has required frontmatter fields
    - min_words: 300
      label: Needs more content depth
    - pattern: '```'
      min: 1
      label: Has code examples
    - has_questions: true
      label: Needs questions for AI search visibility
    - has_answer: true
      label: Needs answer summary for AI citation
questions:
  - How do I build a PTB with the Sui CLI?
  - How do I quote type arguments and object IDs in sui client ptb?
  - How do I deposit and place a DeepBook order in one transaction?
  - How do I write a DeepBook flash loan PTB?
answer: >-
  Build DeepBook programmable transaction blocks from copy-paste templates. The
  sui client ptb CLI uses @-prefixed object IDs, angle-bracket type arguments,
  and quoted arrays. The TypeScript SDK builds the same flows, including deposit
  plus place order in one transaction and flash loan borrow and repay.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> Build common DeepBook [programmable transaction blocks (PTBs)](/develop/transactions/ptbs/prog-txn-blocks) from copy-paste templates for both the [`sui client ptb` CLI](/references/cli/ptb) and the TypeScript SDK. These templates start with the syntax rules that cause the most CLI errors, then cover coin operations, generic Move calls, and DeepBook flows.

Reviewer: ```suggestion
Build common DeepBook [programmable transaction blocks (PTBs)](/develop/transactions/ptbs/prog-txn-blocks) from reusable templates for both the [`sui client ptb` CLI](/references/cli/ptb) and the TypeScript SDK. These templates start with the syntax rules that cause the most CLI errors, then cover coin operations, generic Move calls, and DeepBook flows.
```

CLI examples are verified against `sui` version 1.62.0. DeepBook Move argument order is verified against the [DeepBookV3 sources](https://github.com/MystenLabs/deepbookv3/tree/main/packages/deepbook/sources). Generic TypeScript samples are compile-checked against `@mysten/sui` version 2.22.1. The DeepBook SDK samples follow the documented [DeepBookV3 SDK](/onchain-finance/deepbook/deepbookv3-sdk) usage. None are executed in this document, so dry-run each command before you rely on it.

:::tip

Use `sui client ptb --preview` to print the PTB commands without executing, and `--dry-run` to estimate gas and catch errors before you sign.

:::

> ## Fix the 3 most common CLI errors first

Reviewer: ```suggestion
## Most common `sui client ptb` errors
```
